tui-scrollview/src/scroll_view.rs

```diff
@@ -142,8 +180,8 @@ impl StatefulWidget for ScrollView {
 }
 
 impl ScrollView {
-    /// Render the horizontal and vertical scrollbars if exists, and return the size taken by the
-    /// scrollbars
+    /// Render the horizontal and vertical scrollbars if exist and not hidden,
+    /// and return the size taken by the scrollbars.
     fn render_scrollbars(&self, area: Rect, buf: &mut Buffer, state: &mut ScrollViewState) -> Rect {
```

Comment:
I don't like that this method doubled in size. It makes it harder to reason about and then maintain later.
I suspect that a good split of responsibilities would be:
- working out whether to render each scrollbar
- working out how large each is
- actually rendering the bars

Any thoughts on this?

Reply:
Yeah, I tend to agree. Even before it was not super straightforward to understand where each case would fall, I have to say. I can try and split it (almost did before I sent the PR). Let me know if you have more feedback elsewhere too.

Reply:
Go for it - I'm a big fan of simple obvious code. This is well covered by tests, so refactoring is absolutely fine and the right thing to do. Subjectively my view is that functions should always be less than a screen unless they're big simple homogenous match statements, and that complex nesting is for birds not for software.

It's possible that if the way the code is split makes working out whether to show / how much space they take up something that can happen before the rendering, then this function does not need to return a value with the size taken.